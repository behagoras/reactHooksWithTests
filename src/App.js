import React from 'react'
import {
  BrowserRouter as Router,
  Switch,
  Route,
  Link,
} from 'react-router-dom'
import CounterApp from './components/01-useState/CounterApp'
import CounterWithCustomHook from './components/01-useState/CounterWithCustomHook'

export default function BasicExample() {
  return (
    <Router>
      <>
        <Switch>
          <Route exact path="/">
            <h2>Navigate to exercises</h2>
          </Route>
          <Route path="/counterApp">
            <CounterApp />
          </Route>
          <Route path="/counterWithCustomHook">
            <CounterWithCustomHook />
          </Route>
        </Switch>

        <hr />
        <ul>
          <li>
            <Link to="/">Home</Link>
          </li>
          <li>
            <Link to="/counterApp">CounterApp</Link>
          </li>
          <li>
            <Link to="/counterWithCustomHook">CounterWithCustomHook</Link>
          </li>
        </ul>
      </>
    </Router>
  )
}
