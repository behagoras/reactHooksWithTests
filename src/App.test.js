import { shallow } from 'enzyme'
import React from 'react'
import App from './App'

describe('<App /> ', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<App />)
    expect(wrapper).toMatchSnapshot()
  })
})
