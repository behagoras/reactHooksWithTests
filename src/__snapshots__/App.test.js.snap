// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App />  should render correctly 1`] = `
<div
  className="App"
>
  <BrowserRouter>
    <Menu />
    <hr />
    <Switch>
      <Route
        exact={true}
        path="/"
      >
        <component />
      </Route>
      <Route
        exact={false}
        path="/counterApp"
      >
        <CounterApp />
      </Route>
      <Route
        exact={false}
        path="/counterWithCustomHook"
      >
        <CounterWithCustomHook />
      </Route>
      <Route
        exact={false}
        path="/simpleForm"
      >
        <SimpleForm />
      </Route>
      <Route
        exact={false}
        path="/formWithCustomHook"
      >
        <FormWithCustomHook />
      </Route>
      <Route
        exact={false}
        path="/multipleCustomHooks"
      >
        <MultipleCustomHooks />
      </Route>
      <Route
        exact={false}
        path="/focusScreen"
      >
        <FocusScreen />
      </Route>
      <Route
        exact={false}
        path="/RealExampleRef"
      >
        <RealExampleRef />
      </Route>
      <Route
        exact={false}
        path="/Layout"
      >
        <Layout />
      </Route>
      <Route
        exact={false}
        path="/Memorize"
      >
        <Memorize />
      </Route>
      <Route
        exact={false}
        path="/MemoHook"
      >
        <MemoHook />
      </Route>
      <Route
        exact={false}
        path="/Parent"
      >
        <Memo() />
      </Route>
      <Route
        exact={false}
        path="/ToDoApp"
      >
        <ToDoApp />
      </Route>
      <Route
        exact={false}
        path="/reducer"
      >
        <MainApp />
      </Route>
      <Redirect
        to="/"
      />
    </Switch>
    <hr />
  </BrowserRouter>
</div>
`;
